packages/e2e/test/wallet/SingleAddressWallet/delegation.test.ts

```diff
@@ -67,31 +58,45 @@ const waitForTx = async (wallet: ObservableWallet, { hash }: Transaction.TxInter
 };
 
 describe('SingleAddressWallet/delegation', () => {
-  let wallet1: ObservableWallet;
-  let wallet2: ObservableWallet;
+  let wallet1: Awaited<ReturnType<typeof getWallet>>;
+  let wallet2: Awaited<ReturnType<typeof getWallet>>;
 
   beforeAll(async () => {
     jest.setTimeout(180_000);
     wallet1 = await getWallet({ env, idx: 0, name: 'Test Wallet 1' });
     wallet2 = await getWallet({ env, idx: 1, name: 'Test Wallet 2' });
 
-    await Promise.all([waitForWalletStateSettle(wallet1), waitForWalletStateSettle(wallet2)]);
+    await Promise.all([waitForWalletStateSettle(wallet1.wallet), waitForWalletStateSettle(wallet2.wallet)]);
   });
 
   afterAll(() => {
-    wallet1.shutdown();
-    wallet2.shutdown();
+    wallet1.wallet.shutdown();
+    wallet2.wallet.shutdown();
   });
 
   const chooseWallets = async (): Promise<[ObservableWallet, ObservableWallet]> => {
-    const wallet1Balance = await firstValueFrom(wallet1.balance.utxo.available$);
-    const wallet2Balance = await firstValueFrom(wallet2.balance.utxo.available$);
-    return wallet1Balance.coins > wallet2Balance.coins ? [wallet1, wallet2] : [wallet2, wallet1];
+    const wallet1Balance = await firstValueFrom(wallet1.wallet.balance.utxo.available$);
+    const wallet2Balance = await firstValueFrom(wallet2.wallet.balance.utxo.available$);
+    return wallet1Balance.coins > wallet2Balance.coins
+      ? [wallet1.wallet, wallet2.wallet]
+      : [wallet2.wallet, wallet1.wallet];
+  };
+
+  const chooseDifferentPoolIdRandomly = async (
+    delegateeBefore1stTx: Cardano.PoolId | undefined
+  ): Promise<Cardano.PoolId> => {
+    const activePools = await wallet1.providers.stakePoolProvider.queryStakePools({
+      filters: { status: [Cardano.StakePoolStatus.Active] },
+      pagination: { limit: 5, startAt: 0 }
```

Comment:
That should be slightly faster
```suggestion
      pagination: { limit: 2, startAt: 0 }
```

Reply:
Done in https://github.com/input-output-hk/cardano-js-sdk/pull/369/commits/3b886a7a6529b71445e0928d92921d9b732aa2b1